appservice/src/createAppService/AppInsightsCreateStep.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ApplicationInsightsManagementClient } from 'azure-arm-appinsights';
+import { ResourceManagementClient } from 'azure-arm-resource';
+import { Provider, ProviderResourceType } from 'azure-arm-resource/lib/resource/models';
+import { Location } from 'azure-arm-resource/lib/subscription/models';
+import { Progress } from 'vscode';
+import { AzureWizardExecuteStep, createAzureClient } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { nonNullProp, nonNullValue } from '../utils/nonNull';
+import { IAppServiceWizardContext } from './IAppServiceWizardContext';
+
+export class AppInsightsCreateStep extends AzureWizardExecuteStep<IAppServiceWizardContext> {
+    public priority: number = 135;
+
+    public async execute(wizardContext: IAppServiceWizardContext, progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
+        const location: Location = nonNullProp(wizardContext, 'location');
+        const verifyingAppInsightsAvailable: string = localize('verifyingAppInsightsAvailable', 'Verifying that application insights is available for this location...');
+        ext.outputChannel.appendLine(verifyingAppInsightsAvailable);
+        if (await this.appInsightsSupportedInLocation(wizardContext, location)) {
+            const client: ApplicationInsightsManagementClient = createAzureClient(wizardContext, ApplicationInsightsManagementClient);
+            const rgName: string = nonNullValue(wizardContext.newResourceGroupName);
+            const aiName: string = nonNullValue(wizardContext.newApplicationInsightsName);
+            try {
+                wizardContext.applicationInsights = await client.components.get(rgName, aiName);
+                ext.outputChannel.appendLine(localize('existingNewAppInsights', 'Using existing application insights component "{0}".', aiName));
+            } catch (error) {
+                // Only expecting a resource not found error here
+                const creatingNewAppInsights: string = localize('creatingNewAppInsightsInsights', 'Creating new application insights component "{0}"...', wizardContext.newSiteName);
```

Comment:
You should leverage parseError to check the error type. There's plenty of other errors unrelated to "resource not found" that could happen